### PR TITLE
Zkevm txpool opt

### PR DESCRIPTION
**Pull request by xzav3r**
1. FIX: overflowZkCounter txs must be deleted before `onSenderStateChange `, so we can re-tag them and then promote
2. PERF: do `EnforceBestInvariants()` when peek transactions to improve performance. Yield more transactions each round to prevent calling `EnforceBestInvariants` many times
3. FEAT: do not accept new transactions if pending pool is full. It is better than discard it without any notification

### CODE REVIEW

File: zk/stages/stage_sequence_execute_utils.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ const (
 
 	transactionGasLimit = 30000000
 
-	yieldSize = 100 // arbitrary number defining how many transactions to yield from the pool at once
+	yieldSize = 1000 // arbitrary number defining how many transactions to yield from the pool at once
```
Comment (mandrigin) [inline zk/stages/stage_sequence_execute_utils.go]: let's do it a cli flag, with the default value of 1000

**Review by mandrigin — CHANGES_REQUESTED** [review]
some changes needed

File: zk/stages/stage_sequence_execute_utils.go (hunk 1 above)
Comment (xzav3r) [reply zk/stages/stage_sequence_execute_utils.go]: done

**Comment by V-Staykov** [comment]
Sorry for taking so long to merge this. We need the flag, but we made some other changes as well. Coul'd you merge from zkevm and resolve the conflicts, so we can merge this?

**Comment by xzav3r** [comment]
> Sorry for taking so long to merge this. We need the flag, but we made some other changes as well. Coul'd you merge from zkevm and resolve the conflicts, so we can merge this?

done for that

**Review by V-Staykov — DISMISSED** [review]

**Review by mandrigin — DISMISSED** [review]

**Comment by mandrigin** [comment]
> The base branch requires all commits to be signed. [Learn more about signing commits.](https://docs.github.com/authentication/managing-commit-signature-verification/about-commit-signature-verification)

@yann-sjtu we really want it to be merged, but your commits needs to be verified

**Comment by xzav3r** [comment]
ok, will do

**Comment by xzav3r** [comment]
@mandrigin hi, I force push the code to sign every commits, but no code was changed. Please review again and merge that

**Review by mandrigin — APPROVED** [review]

**Pull request merged**